qiskit_experiments/data_processing/nodes.py

```diff
@@ -341,6 +366,125 @@ def _process(self, datum: np.array, error: Optional[np.array] = None) -> np.arra
+        """Require memory to be a 2D array."""
+        return 2
+
+    def _process(self, datum: np.array) -> np.array:
+        """Take the imaginary part of the IQ data.
+
+        Args:
+            datum: A 2D array of qubits, and a complex averaged IQ point as [real, imaginary].
+
+        Returns:
+            A 1D array. Each entry is the imaginary part of the averaged IQ data of a qubit.
+        """
+        if self.scale is None:
+            return datum[:, 1]
+
+        return datum[:, 1] * self.scale, None
+
+
+class BaseDiscriminator(IQPart):
+    """Base class for discriminator processor. Takes IQ data and calibrated discriminator as input,
+    outputs counts."""
+
+    def __init__(
+        self, handle: ExperimentData, scale: Optional[float] = None, validate: bool = True
+    ):
+        """Initialize a counts to probability data conversion.
+
+        Args:
+            outcome: The bitstring for which to compute the probability.
+            validate: If set to False the DataAction will not validate its input.
+        """
+        self._handle = handle
+        self._scale = scale
+        super().__init__(validate)
+
+    def _required_dimension(self):
+        """Require memory to be a 3D array."""
+        return 3
+
+    def _to_dict(self, list_data: List[int]) -> Dict[str, Any]:
+        """Converts discriminated data in lists to dictionary of counts.
+        Args:
+            list_data: Data in list form at the output of a discriminator.
+
+        Returns:
+            processed data: A dict with the populations.
+        """
+        datum = {}
+        for shot in zip(*list_data):
+            bitstring = "".join(map(str, shot))
+            if bitstring in datum:
+                datum[bitstring] += 1
+            else:
+                datum[bitstring] = 1
+        return datum
+
+    def _process(self, datum: np.array, error: Optional[np.array] = None) -> np.array:
+        # Check that number of qubits are the same between data and discriminator
+        if len(self._handle.analysis_result(0)["discriminator"]) != np.shape(datum)[1]:
+            raise DataProcessorError(
+                "The number of qubits of data and discriminator must be the same."
+            )
+
+
+class LDADiscriminator(BaseDiscriminator):
+    def _process(self, datum: np.array, error: Optional[np.array] = None) -> np.array:
+        """Applies LDA discriminator to IQ data to return counts.
+        Args:
+            datum: Input IQ data to be discriminated.
+
+        Returns:
+            processed data: Counts dictionary.
+        """
+        super()._process(datum)
+
+        list_data = []
+        for i in range(np.shape(datum)[1]):
+            if (
+                isinstance(
+                    self._handle.analysis_result(0)["discriminator"][i], LinearDiscriminantAnalysis
+                )
+                is False
+            ):
+                raise DataProcessorError("Input not an LDA discriminator.")
```

Comment:
Do we need this? It looks like you could just have one type of discriminator node.